pysmt/fnode.py

```diff
@@ -180,6 +196,91 @@ def is_le(self):
     def is_lt(self):
         return self.node_type() == LT
 
+    def is_bv_not(self):
+        return self.node_type() == BV_NOT
+
+    def is_bv_and(self):
+        return self.node_type() == BV_AND
+
+    def is_bv_or(self):
+        return self.node_type() == BV_OR
+
+    def is_bv_xor(self):
+        return self.node_type() == BV_XOR
+
+    def is_bv_concat(self):
+        return self.node_type() == BV_CONCAT
+
+    def is_bv_extract(self):
+        return self.node_type() == BV_EXTRACT
+
+    def is_bv_ult(self):
+        return self.node_type() == BV_ULT
+
+    def is_bv_neg(self):
+        return self.node_type() == BV_NEG
+
+    def is_bv_add(self):
+        return self.node_type() == BV_ADD
+
+    def is_bv_mul(self):
+        return self.node_type() == BV_MUL
+
+    def is_bv_udiv(self):
+        return self.node_type() == BV_UDIV
+
+    def is_bv_urem(self):
+        return self.node_type() == BV_UREM
+
+    def is_bv_lshl(self):
+        return self.node_type() == BV_LSHL
+
+    def is_bv_lshr(self):
+        return self.node_type() == BV_LSHR
+
+    def is_bv_rol(self):
+        return self.node_type() == BV_ROL
+
+    def is_bv_ror(self):
+        return self.node_type() == BV_ROR
+
+    def is_bv_zext(self):
+        return self.node_type() == BV_ZEXT
+
+    def is_bv_sext(self):
+        return self.node_type() == BV_SEXT
+
+    def bv_width(self):
+        if self.is_bv_constant():
+            return self._content.payload[1]
+        elif self.is_symbol():
+            assert self.symbol_type().is_bv_type()
+            return self.symbol_type().width
+        elif self.is_function_application():
+            # Return width defined in the declaration
+            return self.function_name().symbol_type().return_type.width
+        elif self.is_ite():
+            # Recursively call bv_width on the children
+            width_l = self.arg(1).bv_width()
```

Comment:
We could cache this. Moreover, it is probably not necessary to check everytime that the width is the same, since this should be checked at build time.